### ISSUE TITLE
Right aligned elements font scaling problem

### ISSUE THREAD

**Issue by kirsan31** [dotnet/winforms#5076]
* .NET Core Version: all up to latest tested 5.0.6

* Have you experienced this same bug with .NET Framework?: No (not applicable).

**Problem description:**

Right aligned elements on the form can change their relative position after font scaling.
For example, when you design a form in designer you often right align labels before input fields:
![image](https://user-images.githubusercontent.com/17767561/121644245-75cc5880-ca9b-11eb-852c-435bd0515f9e.png)

But after you ran this app in .Net, you will see this:
![image](https://user-images.githubusercontent.com/17767561/121644713-099e2480-ca9c-11eb-876b-6ec8e0b99874.png)
Right align is broken due to Font scaling (default font change in net core).

Most interesting things, same app above:
![image](https://user-images.githubusercontent.com/17767561/121645071-74e7f680-ca9c-11eb-90a4-bc8364166e70.png)
![image](https://user-images.githubusercontent.com/17767561/121645261-a82a8580-ca9c-11eb-8f8a-b7ae6931d564.png)
And Right align is ok. ~~I really don't know how this happens :) It seems that's it bugs only with 100% dpi and 100% text size. But this statement must be double checked...~~ Seems it's all depends on the final font size.

Because we are using old designer here it's seems that this is only migration issue. But this is not true, due to new designer is not ready for production use, and we are still force to use old one with .Net :(

**Expected behavior:**

Right align is not broken.

**Minimal repro:**

[RightAlign.zip](https://github.com/dotnet/winforms/files/6636505/RightAlign.zip)

**Comment by dreddy-work** [comment]
@kirsan31 , you don't see this when AutoscaleMode  is DPI?

**Comment by kirsan31** [comment]
@dreddy-work 
> you don't see this when AutoscaleMode is DPI?

Yea, with AutoscaleMode = DPI align is ok, and it's logical:
![image](https://user-images.githubusercontent.com/17767561/121745110-a13b5b80-cb0c-11eb-9f37-2707c62672eb.png)

**Comment by dreddy-work** [comment]
In this case, i do not think winforms has any control over it. We rely on Font metrics when scaling and they come from windows. Users have alternative to set DPI scale. When it comes to high DPI, our choice should default DPI scale but that would be a breaking change from current default Font scale.

In Core, Designer is rendered as out of proc and use the targeted runtime. So, it may be worth checking the behavior there.

**Comment by kirsan31** [comment]
@dreddy-work 
Hmm, and if we look to this from other side?
Open this project in new designer (see wrong align), correct right align, build, change dpi from 100 to 125, run app - align is wrong:

https://user-images.githubusercontent.com/17767561/121750611-5eca4c80-cb15-11eb-8877-30e55ec1b9eb.mp4

P.s. We relay on Font scaling instead of DPI scaling because our apps are all accessed only by rdp. And changing dpi on the server from rdp session is still challenging now days (especially with all park of different rdp clients) :( But Font scaling is work easy in this case.

**Comment by dreddy-work** [comment]
Thanks @kirsan31 for the GIF demonstrating the issue. A logical fix of right aligning the label will still create same alignment issue with controls on the left. Label size is adjusted to the text in it no matter if it is DPI ( DPI change cause Font change) or Font scaled.

**Comment by kirsan31** [comment]
@dreddy-work did you got that with `AutoscaleMode = DPI` issue is still present (picture after video)?

Overall I understand that to scale well right (bottom?) aligned controls we need a new technique here. Control need to know from what corner it's must be scaled.

Can I close this or we leave this for distant future? :-)

**Comment by dreddy-work** [comment]
Yes. for RTL, we need special handling on the DPI changed.  I have not dig into it yet but keep this bug open. We will update it with info and decide then if this need to be closed.

**Comment by Olina-Zhang** [comment]
@dreddy-work  Verified this issue in the latest .Net 7.0 SDK build, this issue doesn't reproduce now. When Right aligned controls on 100% main screen, the runtime has the same behavior as design time when changing DPI to 300% main screen. Please see following screenshot on 300% main screen:
![image](https://user-images.githubusercontent.com/26474449/142988440-9333825c-53bc-43f8-ab6e-b24e287ffd54.png)

**Comment by RussKie** [comment]
Presume fixed. @kirsan31 please let us know if the issue resurfaces.

**Comment by kirsan31** [comment]
@RussKie with `7.0.100-alpha.1.21568.2` all problems described in [first](https://github.com/dotnet/winforms/issues/5076#issue-918325809) and [fifth](https://github.com/dotnet/winforms/issues/5076#issuecomment-859923019) posts are persists.
